From 9ce7deff92e89b1807d0968c0b319d26361a72a6 Mon Sep 17 00:00:00 2001
Message-Id: <9ce7deff92e89b1807d0968c0b319d26361a72a6.1654484531.git.geliang.tang@suse.com>
From: Mat Martineau <mathew.j.martineau@linux.intel.com>
Date: Tue, 3 Nov 2020 11:05:08 -0800
Subject: [PATCH] docs: networking: mptcp: Add MPTCP sysctl entries
Git-commit: 9ce7deff92e89b1807d0968c0b319d26361a72a6
Patch-mainline: v5.11-rc1
References: bsn#2
Modified-by-SEL: No

Describe the two MPTCP sysctls, what the values mean, and the default
settings.

Acked-by: Geliang Tang <geliangtang@gmail.com>
Signed-off-by: Mat Martineau <mathew.j.martineau@linux.intel.com>
Reviewed-by: Matthieu Baerts <matthieu.baerts@tessares.net>
Signed-off-by: Jakub Kicinski <kuba@kernel.org>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 Documentation/networking/index.rst        |  1 +
 Documentation/networking/mptcp-sysctl.rst | 26 +++++++++++++++++++++++
 MAINTAINERS                               |  1 +
 3 files changed, 28 insertions(+)
 create mode 100644 Documentation/networking/mptcp-sysctl.rst

diff --git a/Documentation/networking/index.rst b/Documentation/networking/index.rst
index 63ef386afd0a..70c71c9206e2 100644
--- a/Documentation/networking/index.rst
+++ b/Documentation/networking/index.rst
@@ -70,6 +70,7 @@ Contents:
    lapb-module
    mac80211-injection
    mpls-sysctl
+   mptcp-sysctl
    multiqueue
    netconsole
    netdev-features
diff --git a/Documentation/networking/mptcp-sysctl.rst b/Documentation/networking/mptcp-sysctl.rst
new file mode 100644
index 000000000000..6af0196c4297
--- /dev/null
+++ b/Documentation/networking/mptcp-sysctl.rst
@@ -0,0 +1,26 @@
+.. SPDX-License-Identifier: GPL-2.0
+
+=====================
+MPTCP Sysfs variables
+=====================
+
+/proc/sys/net/mptcp/* Variables
+===============================
+
+enabled - INTEGER
+	Control whether MPTCP sockets can be created.
+
+	MPTCP sockets can be created if the value is nonzero. This is
+	a per-namespace sysctl.
+
+	Default: 1
+
+add_addr_timeout - INTEGER (seconds)
+	Set the timeout after which an ADD_ADDR control message will be
+	resent to an MPTCP peer that has not acknowledged a previous
+	ADD_ADDR message.
+
+	The default value matches TCP_RTO_MAX. This is a per-namespace
+	sysctl.
+
+	Default: 120
diff --git a/MAINTAINERS b/MAINTAINERS
index 17f5571788c9..badaaa815aa3 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -12265,6 +12265,7 @@ L:	mptcp@lists.01.org
 S:	Maintained
 W:	https://github.com/multipath-tcp/mptcp_net-next/wiki
 B:	https://github.com/multipath-tcp/mptcp_net-next/issues
+F:	Documentation/networking/mptcp-sysctl.rst
 F:	include/net/mptcp.h
 F:	include/uapi/linux/mptcp.h
 F:	net/mptcp/
-- 
2.34.1

